source/utils/CMakeLists.txt

```diff
@@ -12,6 +12,9 @@ configure_file(
 # BPLS
 add_executable(bpls ./bpls/bpls.cpp)
 target_link_libraries(bpls adios2_core adios2sys adios2::thirdparty::pugixml)
+if(WIN32)
+  target_link_libraries(bpls shlwapi)
+endif()
```

Comment:
```suggestion
target_link_libraries(bpls PRIVATE $<$<PLATFORM_ID:Windows>:shlwapi>)
```

Reply:
Should I mention "PRIVATE" keyword? It can not be used due to this https://stackoverflow.com/a/59522268

Reply:
Good catch, perhaps adding `PUBLIC` for in the first call would be a workaround. @Biswa96 normally we want to make dependencies private unless otherwise.

> Also what is the problem with the current change?

There is nothing wrong, this is a code suggestion within a code review.